src/Stratis.Bitcoin/Primitives/ChainedHeaderBlock.cs

```diff
@@ -32,5 +33,17 @@ public override string ToString()
         {
             return this.ChainedHeader.ToString();
         }
+
+        /// <summary>
+        /// Update the <see cref="ChainedHeader" /> (if not null) with a new provided header.
+        /// </summary>
+        /// <param name="newHeader">The new header to set.</param>
+        /// <remarks>Use this method very carefully because it could cause race conditions if used at the wrong moment.</remarks>
+        public void SetHeader(BlockHeader newHeader)
+        {
+            Guard.NotNull(newHeader, nameof(newHeader));
+
+            this.ChainedHeader?.SetHeader(newHeader);
```

Comment:
This is a problematic method because we dont know for sure if header was set, we are better of doing `chainedHeaderBlock.ChainedHeader.SetBlock()`

Reply:
Ah actually ChainedHeader can never be null

Reply:
that method is inside ChainedHeaderBlock and just ensure that ChainedHeader is set, with `?.` preventing to invoke SetBlock if ChainedHeader is null (even if this should never happen, right?)

maybe I don't understand, why do you want to call `SetBlock` if we are setting an header, typo?